presto-main/src/test/java/com/facebook/presto/type/TestRealOperators.java

```diff
@@ -245,4 +248,13 @@ public void testIndeterminate()
         assertOperator(INDETERMINATE, "cast(1.2 as real)", BOOLEAN, false);
         assertOperator(INDETERMINATE, "cast(123 as real)", BOOLEAN, false);
     }
+
+    @Test
+    public void testNanHash()
```

Comment:
can you name this method `testHash` and add some non-NaN cases as well?

can you also add similar test in `TestDoubleOperators` ?

Reply:
I am not sure how are we going to test non-Nan cases. It's hard to define correct behavior of `hash()`. We can check if there is no collision but that's not very informative. 
 
Do we want the output to be exactly the same as `REAL.hash()` which is a block+offset version defined in `RealType`? Or any other ideas?

Also, for `DoubleOperator`, it's native double type so we don't need to worry about `NaN`.